macosx/VersionComparator.mm

```diff
@@ -0,0 +1,43 @@
+// This file Copyright © 2023 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "VersionComparator.h"
+
+@implementation VersionComparator
+- (NSComparisonResult)compareVersion:(NSString*)versionA toVersion:(NSString*)versionB
+{
+    // Transmission version format follows:
+    // 14714.major.minor.patch.beta
+    // caveat: "beta" is a pre-release number, so it comes before a release version.
+    NSArray<NSString*>* versionBComponents = [versionB componentsSeparatedByString:@"."];
+    if (versionBComponents.count > 4 && versionBComponents[4].integerValue > 0 &&
+        ![NSUserDefaults.standardUserDefaults boolForKey:@"AutoUpdateBeta"] &&
+        ![[[NSBundle mainBundle] objectForInfoDictionaryKey:(NSString*)kCFBundleVersionKey] isEqualToString:versionB])
+    {
+        // pre-releases are ignored
+        return NSOrderedDescending;
+    }
+    NSArray<NSString*>* versionAComponents = [versionA componentsSeparatedByString:@"."];
+    for (NSUInteger idx = 0; versionAComponents.count > idx || versionBComponents.count > idx; idx++)
```

Comment:
you can swap in `MAX(versionAComponents.count, versionBComponents.count) > idx`

Reply:
I'm a bit happier without using a macro here.
I get it that we save 3 characters, but under the hood, expanding a macro makes it more complex to review.

Reply:
Marking as resolved because this code has been removed from the PR